### ISSUE TITLE
[BUG]: Being in a different team than expected makes the particles get indexed out of bounds

### ISSUE THREAD

**Issue by StShAzzy** [Phil25/RTD#122]
check the player's team if it's unassigned or spectator, and if it is, just force to use either red or blue variant

or if you're that mean, you could just not allow players to roll if they aren't on either red or blue team, which would suck as it's not the proper fix

**Comment by StShAzzy** [comment]
https://github.com/Phil25/RTD/blob/26aba4fc19b7390fd1e4a8b341bb25abfae4e068/scripting/rtd/perks/autoheal.sp#L39

this one for instance when you're in spectator but alive and players (has a playermodel and everything else, but it's just in another team)
causes the particles to get indexed out of bounds, grabbing any junk nearby, some times it's harmless because the particles despawn, other times, spawns particles that don't despawn, which means the client will crash if too many spawn, as it runs out of edicts for rendering particles

**Comment by Phil25** [comment]
I don't recall spectator abuse being an issue, so not sure why these problems are coming out now. I'm wondering if anything changed in the latest upgrades, maybe the 64 bit one, where players in spectators are considered alive.

In any case, RTD hasn't been designed with this in mind, so that's definitely needing of a fix.

**Comment by StShAzzy** [comment]
no, you can just forcefully set players to different teams using plugins

**Comment by StShAzzy** [comment]
without using the SourceMod function that also kills you, by changing `m_iTeamNum` to 0 or 1

**Comment by StShAzzy** [comment]
if any question arise in your head regarding this, please contact me, i'll be happy to make sure it gets fixed properly

**Comment by StShAzzy** [comment]
> I don't recall spectator abuse being an issue, so not sure why these problems are coming out now. I'm wondering if anything changed in the latest upgrades, maybe the 64 bit one, where players in spectators are considered alive.


i was just too lazy to actually come and report it as soon as i found it out and could replicate it
